### ISSUE TITLE
Play more nicely with an existing Orbot install

### ISSUE THREAD

**Issue by geebotron** [horizontalsystems/unstoppable-wallet-android#3959]
On android UW seems upset if there is an existing instance of orbot running. Stopping orbot, restarting UW and then restarting orbot seems to fix this. (I think orbot will try an alternative port to 9050, if another process is already using it.)

Perhaps an option to use orbot for the Tor connection, rather than the UW built in connector?

Thanks.

**Comment by abdrasulov** [comment]
Hi @geebotron 

We couldn't reproduce the issue. Can you please share the details? What is your device and android version? What the app behavior in this case? Does it stop syncing with blockchains?

**Comment by geebotron** [comment]
My phone is a Samsung S21+ running Android 11.

To reproduce:
1. Configure UW to use its built in version of Tor.
2. Kill both UW and Orbot.
3. Start Orbot. It will by default use port 9050 for its proxy interface.
4. Start UW. After requesting the passcode it hangs with a message: "Error starting Tor!", with options to retry or disable Tor. I assume this is because it is hard-coded to use port 9050?

To fix:
1. Kill both UW and Orbot
2. Start UW - after requesting passcode it connects via Tor.
3. Start Orbot - it starts and uses port 9051 for its proxy interface, rather than 9050 - which I assume Orbot has noticed is already being used by UW

I have also observed that apps (like F-Droid) that have an actual "use Orbot" config option (rather than just a fixed socks5 proxy host/port config) can deal with Orbot running on an unusual port, other than 9050, so I guess they query the table of processes to find the orbot process and subsequently detect the port being used by the orbot process.

In other words - even if you kill the UW process (freeing up port 9050) but do NOT restart Orbot (so it continues to use some other port e.g. 9051) apps with a "proper config" for Orbot, continue to work - making Tor connections via Orbot.

**Comment by abdrasulov** [comment]
@geebotron Thanks for clarifications! We'll work on it in the upcoming releases.

**Comment by esengulov** [comment]
@abdrasulov please look into this and try to apply recommended fix by @geebotron . I f it works then great. don't put too much time on it.
